src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -1328,7 +1328,8 @@ public SolrInputDocuments toSolrDocs(IndexableDataset indexableDataset, Set<Long
             }
             LocalDate embargoEndDate=null;
             LocalDate retentionEndDate=null;
-            final String datasetCitation = dataset.getCitation();
+            final String datasetCitation = (dataset.isReleased() && dataset.getReleasedVersion() != null) ?
```

Comment:
You wouldn't ever expect that a released dataset wouldn't have a released version unless maybe there's a deaccessioned version, maybe?